frostsnap_coordinator/src/bitcoin/wallet.rs

```diff
@@ -760,10 +778,82 @@ impl CoordSuperWallet {
             }
         }
 
+        // Validate that this PSBT is actually signable by this wallet
+        if owned_count == 0 {
+            return Err(PsbtValidationError::NothingToSign {
+                total_inputs: psbt.inputs.len(),
+                foreign_count,
+                already_signed_count,
+            });
+        }
+
         Ok(template)
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum PsbtValidationError {
+    NothingToSign {
+        total_inputs: usize,
+        foreign_count: usize,
+        already_signed_count: usize,
+    },
+    Other(String),
+}
+
+impl From<Box<bitcoin_transaction::SpkDoesntMatchPathError>> for PsbtValidationError {
+    fn from(e: Box<bitcoin_transaction::SpkDoesntMatchPathError>) -> Self {
+        PsbtValidationError::Other(e.to_string())
+    }
+}
+
+impl From<anyhow::Error> for PsbtValidationError {
+    fn from(e: anyhow::Error) -> Self {
+        PsbtValidationError::Other(e.to_string())
+    }
+}
+
+impl std::fmt::Display for PsbtValidationError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PsbtValidationError::NothingToSign {
+                total_inputs,
+                foreign_count,
+                already_signed_count,
+            } => {
+                let input_word = if *total_inputs == 1 {
+                    "input"
+                } else {
+                    "inputs"
+                };
+                write!(
+                    f,
+                    "This PSBT has {} {} but this wallet can not sign any of them",
+                    total_inputs, input_word
+                )?;
+
+                if *foreign_count > 0 && *already_signed_count > 0 {
+                    write!(
+                        f,
+                        " ({} not owned by this wallet, {} already signed)",
+                        foreign_count, already_signed_count
+                    )?;
+                } else if *foreign_count > 0 {
+                    write!(f, " ({} not owned by this wallet)", foreign_count)?;
```

Comment:
Could be worth stating that frostsnap does not yet support signing of collaborative transactions, but not sure if here is the right place